packages/ra-core/src/controller/ListController.spec.tsx

```diff
@@ -113,6 +118,23 @@ describe('ListController', () => {
             expect(props.changeListParams.mock.calls[0][1].filter).toEqual({});
         });
 
+        it('should update data if permanent filters change', () => {
+            expect.assertions(2);
+            const props = {
+                ...defaultProps,
+                debounce: 200,
+                crudGetList: jest.fn(),
+                filter: { foo: 1 },
+                children: fakeComponent,
+            };
+
+            const wrapper = shallow(<ListController {...props} />);
+            expect(props.crudGetList.mock.calls[0][3]).toEqual({ foo: 1 });
+
+            wrapper.setProps({ filter: { foo: 2 } });
+            expect(props.crudGetList.mock.calls[1][3]).toEqual({ foo: 2 });
+        });
+
```

Comment:
Could you also add a test that permanent filters don't appear in the filter form and button?

Reply:
bump

Reply:
Done

Reply:
Added comments on the tests to explain